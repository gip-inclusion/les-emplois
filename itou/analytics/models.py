import uuid

from django.db import models
from django.utils import timezone

from itou.common_apps.address.departments import DEPARTMENTS
from itou.users.enums import UserKind


class DatumCode(models.TextChoices):
    # Employee record - Base
    EMPLOYEE_RECORD_COUNT = "ER-001", "FS totales"
    EMPLOYEE_RECORD_DELETED = "ER-002", "FS (probablement) supprimées"
    # Employee record - Lifecycle
    EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE = "ER-101", "FS intégrées (0000) au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE = "ER-102", "FS avec une erreur au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE = "ER-102-3436", "FS avec une erreur 3436 au premier retour"
    EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR = "ER-103", "FS ayant eu au moins un retour en erreur"
    # Approval - Base
    APPROVAL_COUNT = "AP-001", "PASS IAE total"
    APPROVAL_CANCELLED = "AP-002", "PASS IAE annulés"
    # Approval - PE notification cycle
    APPROVAL_PE_NOTIFY_SUCCESS = "AP-101", "PASS IAE synchronisés avec succès avec pole emploi"
    APPROVAL_PE_NOTIFY_PENDING = "AP-102", "PASS IAE en attente de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_ERROR = "AP-103", "PASS IAE en erreur de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_READY = "AP-104", "PASS IAE prêts à être synchronisés avec pole emploi"
    # Users
    USER_COUNT = "US-001", "Nombre d'utilisateurs"
    USER_JOB_SEEKER_COUNT = "US-011", "Nombre de demandeurs d'emploi"
    USER_PRESCRIBER_COUNT = "US-012", "Nombre de prescripteurs"
    USER_EMPLOYER_COUNT = "US-013", "Nombre d'employeurs"
    USER_LABOR_INSPECTOR_COUNT = "US-014", "Nombre d'inspecteurs du travail"
    USER_ITOU_STAFF_COUNT = "US-015", "Nombre d'administrateurs"
    # API usage
    API_TOTAL_CALLS = "API-001", "API : total d'appels reçus"
    API_TOTAL_UV = "API-002", "API : total de visiteurs uniques"
    API_TOTAL_CALLS_CANDIDATS = "API-003", "API candidats : total d'appels reçus"
    API_TOTAL_UV_CANDIDATS = "API-004", "API candidats : total de visiteurs uniques"
    API_TOTAL_CALLS_GEIQ = "API-005", "API GEIQ : total d'appels reçus"
    API_TOTAL_UV_GEIQ = "API-006", "API GEIQ : total de visiteurs uniques"
    API_TOTAL_CALLS_ER = "API-007", "API FS : total d'appels reçus"
    API_TOTAL_UV_ER = "API-008", "API FS : total de visiteurs uniques"
    API_TOTAL_CALLS_MARCHE = "API-009", "API Le marché : total d'appels reçus"
    API_TOTAL_CALLS_SIAES = "API-010", "API siaes : total d'appels reçus"
    API_TOTAL_UV_SIAES = "API-011", "API siaes : total de visiteurs uniques (par adresse IP)"
    API_TOTAL_CALLS_STRUCTURES = "API-012", "API structures : total d'appels reçus"


class Datum(models.Model):
    """Store an aggregated `value` of the `code` data point for the specified `bucket`."""

    id = models.UUIDField(primary_key=True, default=uuid.uuid4)

    code = models.TextField(choices=DatumCode.choices)
    bucket = models.TextField()
    value = models.IntegerField()  # Integer offers the best balance between range, storage size, and performance

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "data"
        unique_together = ["code", "bucket"]
        indexes = [models.Index(fields=["measured_at", "code"])]


class StatsDashboardVisit(models.Model):
    dashboard_id = models.IntegerField(verbose_name="ID tableau de bord Metabase")
    dashboard_name = models.TextField(verbose_name="nom de la vue du tableau de bord")
    department = models.CharField(verbose_name="département", choices=DEPARTMENTS.items(), max_length=3, null=True)
    region = models.TextField(verbose_name="région", null=True)
    current_company_id = models.IntegerField(verbose_name="ID entreprise courante", null=True)
    current_prescriber_organization_id = models.IntegerField(
        verbose_name="ID organisation prescriptrice courante", null=True
    )
    current_institution_id = models.IntegerField(verbose_name="ID institution courante", null=True)
    user_kind = models.TextField(verbose_name="type d'utilisateur", choices=UserKind.choices)
    user_id = models.IntegerField(verbose_name="ID utilisateur")

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "visite de tableau de bord"
