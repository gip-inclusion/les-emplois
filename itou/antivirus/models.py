from django.db import models
from django.utils.text import capfirst

from itou.files.models import File


class Scan(models.Model):
    file = models.OneToOneField(File, on_delete=models.CASCADE)
    clamav_signature = models.TextField()
    clamav_completed_at = models.DateTimeField(null=True, verbose_name="analyse ClamAV le")
    clamav_infected = models.BooleanField(null=True, verbose_name="fichier infect√© selon ClamAV")
    comment = models.TextField(blank=True, verbose_name="commentaire")

    class Meta:
        verbose_name = "analyse antivirus"
        verbose_name_plural = "analyses antivirus"

    def __str__(self):
        text = f"{capfirst(self._meta.verbose_name)} {self.file_id}"
        if self.clamav_infected:
            text = f"[VIRUS] {text}"
        return text
