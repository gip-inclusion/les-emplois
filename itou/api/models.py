import secrets

from django.db import models
from django.utils import timezone

from itou.companies.models import Company


def _generate_key():
    return secrets.token_urlsafe()


class CompanyToken(models.Model):
    key = models.CharField(default=_generate_key, unique=True, editable=False)
    label = models.CharField(verbose_name="m√©mo permettant d'identifier l'usage du jeton", max_length=60, unique=True)
    created_at = models.DateTimeField(default=timezone.now)
    companies = models.ManyToManyField(Company, related_name="api_tokens")

    class Meta:
        verbose_name = "jeton d'API entreprise"
        verbose_name_plural = "jetons d'API entreprise"
