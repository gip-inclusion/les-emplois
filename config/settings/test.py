import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
