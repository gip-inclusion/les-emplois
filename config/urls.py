from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
