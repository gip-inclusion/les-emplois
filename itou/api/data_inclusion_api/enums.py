import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
