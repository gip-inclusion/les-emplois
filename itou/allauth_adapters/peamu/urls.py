from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
