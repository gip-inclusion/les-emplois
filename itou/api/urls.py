from django.urls import include, path
from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView
from rest_framework import routers
from rest_framework.authtoken import views as auth_views

from itou.api.data_inclusion_api.views import DataInclusionStructureView

from .applicants_api.views import ApplicantsView
from .employee_record_api.viewsets import (
    DummyEmployeeRecordViewSet,
    EmployeeRecordUpdateNotificationViewSet,
    EmployeeRecordViewSet,
)
from .siae_api.viewsets import SiaeViewSet


# High level app for API
# https://docs.djangoproject.com/en/dev/topics/http/urls/#url-namespaces-and-included-urlconfs
app_name = "itou.api"

# Using DRF router with viewsets means automatic definition of utl patterns
router = routers.DefaultRouter()
router.register(r"employee-records", EmployeeRecordViewSet, basename="employee-records")
router.register(
    r"employee-record-notifications",
    EmployeeRecordUpdateNotificationViewSet,
    basename="employee-record-notifications",
)
router.register(r"dummy-employee-records", DummyEmployeeRecordViewSet, basename="dummy-employee-records")
router.register(r"siaes", SiaeViewSet, basename="siaes")

urlpatterns = [
    # TokenAuthentication endpoint to get token from login/password.
    path("token-auth/", auth_views.obtain_auth_token, name="token-auth"),
    # Needed for Browseable API (dev)
    path("api-auth/", include("rest_framework.urls", namespace="rest_framework")),
    # OpenAPI
    # See: https://www.django-rest-framework.org/topics/documenting-your-api/
    # OAS 3 YAML schema (downloadable)
    path(
        "oas3/",
        SpectacularAPIView.as_view(),
        name="openapi_schema",
    ),
    path(
        "redoc/",
        SpectacularRedocView.as_view(url_name="v1:openapi_schema"),
        name="redoc",
    ),
]

urlpatterns += router.urls

urlpatterns += [
    path("structures/", DataInclusionStructureView.as_view(), name="structures-list"),
    path("candidats/", ApplicantsView.as_view(), name="applicants-list"),
]
