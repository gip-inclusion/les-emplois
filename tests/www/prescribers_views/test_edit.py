from unittest import mock

from django.urls import reverse
from pytest_django.asserts import assertContains

from itou.prescribers.enums import PrescriberOrganizationKind
from itou.prescribers.factories import PrescriberOrganizationFactory, PrescriberOrganizationWithMembershipFactory
from itou.prescribers.models import PrescriberOrganization
from itou.utils.mocks.geocoding import BAN_GEOCODING_API_RESULT_MOCK
from tests.utils.test import TestCase, parse_response_to_soup


class CardViewTest(TestCase):
    def test_card(self):
        prescriber_org = PrescriberOrganizationFactory(is_authorized=True)
        url = reverse("prescribers_views:card", kwargs={"org_id": prescriber_org.pk})
        response = self.client.get(url)
        assert response.status_code == 200
        assert response.context["prescriber_org"] == prescriber_org


class TestEditOrganization:
    def test_edit(self, client):
        """Edit a prescriber organization."""

        organization = PrescriberOrganizationWithMembershipFactory(
            authorized=True, kind=PrescriberOrganizationKind.CAP_EMPLOI
        )
        user = organization.members.first()

        client.force_login(user)

        url = reverse("prescribers_views:edit_organization")
        response = client.get(url)
        assert response.status_code == 200

        post_data = {
            "name": "foo",
            "siret": organization.siret,
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
            "address_line_1": "2 Rue de Soufflenheim",
            "address_line_2": "",
            "city": "Betschdorf",
            "post_code": "67660",
            "department": "67",
            "email": "",
            "phone": "0610203050",
            "website": "https://famous-siae.com",
        }
        with mock.patch(
            "itou.utils.apis.geocoding.call_ban_geocoding_api",
            return_value=BAN_GEOCODING_API_RESULT_MOCK,
        ) as mock_call_ban_geocoding_api:
            response = client.post(url, data=post_data)
        assert response.status_code == 302

        mock_call_ban_geocoding_api.assert_called_once()

        organization = PrescriberOrganization.objects.get(siret=organization.siret)

        assert organization.name == post_data["name"]
        assert organization.description == post_data["description"]
        assert organization.address_line_1 == post_data["address_line_1"]
        assert organization.address_line_2 == post_data["address_line_2"]
        assert organization.city == post_data["city"]
        assert organization.post_code == post_data["post_code"]
        assert organization.department == post_data["department"]
        assert organization.email == post_data["email"]
        assert organization.phone == post_data["phone"]
        assert organization.website == post_data["website"]

        # This data comes from BAN_GEOCODING_API_RESULT_MOCK.
        assert organization.coords == "SRID=4326;POINT (2.316754 48.838411)"
        assert organization.latitude == 48.838411
        assert organization.longitude == 2.316754
        assert organization.geocoding_score == 0.587663373207207

        # Only admins should be able to edit organization details
        membership = organization.prescribermembership_set.first()
        membership.is_admin = False
        membership.save()
        url = reverse("prescribers_views:edit_organization")
        response = client.get(url)
        assert response.status_code == 403

    def test_edit_with_multiple_memberships_and_same_siret(self, client):
        """
        Updating information of the prescriber organization must be possible
        when user is member of multiple orgs with the same SIRET (and different types)
        (was a regression)
        """
        organization = PrescriberOrganizationWithMembershipFactory(kind=PrescriberOrganizationKind.ML)
        siret = organization.siret
        user = organization.members.first()

        org2 = PrescriberOrganizationWithMembershipFactory(kind=PrescriberOrganizationKind.PLIE, siret=siret)
        org2.members.add(user)
        org2.save()

        client.force_login(user)

        url = reverse("prescribers_views:edit_organization")
        response = client.get(url)
        assert response.status_code == 200

        post_data = {
            "siret": siret,
            "name": "foo",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
            "address_line_1": "2 Rue de Soufflenheim",
            "address_line_2": "",
            "city": "Betschdorf",
            "post_code": "67660",
            "department": "67",
            "email": "",
            "phone": "0610203050",
            "website": "https://famous-siae.com",
        }
        with mock.patch(
            "itou.utils.apis.geocoding.call_ban_geocoding_api",
            return_value=BAN_GEOCODING_API_RESULT_MOCK,
        ):
            response = client.post(url, data=post_data)
        assert response.status_code == 302

        url = reverse("dashboard:index")
        assert url == response.url

    def test_pe_cannot_edit(self, client, snapshot):
        organization = PrescriberOrganizationWithMembershipFactory(
            authorized=True,
            kind=PrescriberOrganizationKind.PE,
            name="PÃ´le emploi",
            siret="12345678901234",
            phone="0600000000",
            email="pe@mailinator.com",
            department="53",
        )
        user = organization.members.first()
        url = reverse("prescribers_views:edit_organization")

        client.force_login(user)
        response = client.get(url)
        form = parse_response_to_soup(response, selector="form.js-prevent-multiple-submit")
        assert str(form) == snapshot(name="Fields are disabled")
        assertContains(response, "Affichage des informations en lecture seule", count=1)

        post_data = {
            "name": "foo",
            "siret": organization.siret,
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
            "address_line_1": "2 Rue de Soufflenheim",
            "address_line_2": "",
            "city": "Betschdorf",
            "post_code": "67660",
            "department": "67",
            "email": "",
            "phone": "0610203050",
            "website": "https://famous-siae.com",
        }
        with mock.patch(
            "itou.utils.apis.geocoding.call_ban_geocoding_api",
            return_value=BAN_GEOCODING_API_RESULT_MOCK,
        ) as mock_call_ban_geocoding_api:
            response = client.post(url, data=post_data)
        assert response.status_code == 200
        mock_call_ban_geocoding_api.assert_not_called()

        # Data was not modified.
        organization_refreshed = PrescriberOrganization.objects.get(pk=organization.pk)
        for field in [f for f in PrescriberOrganization._meta.get_fields() if not f.is_relation]:
            assert getattr(organization, field.name) == getattr(organization_refreshed, field.name)
