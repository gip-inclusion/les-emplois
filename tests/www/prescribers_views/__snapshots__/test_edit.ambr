# serializer version: 1
# name: TestEditOrganization.test_pe_cannot_edit[Fields are disabled]
  '''
  <form class="js-prevent-multiple-submit" method="post">
  
                              <input name="csrfmiddlewaretoken" type="hidden" value="NORMALIZED_CSRF_TOKEN"/>
  
                              <div class="form-group form-group-required"><label class="form-label" for="id_address_line_1">Adresse</label><input class="form-control" disabled="" id="id_address_line_1" maxlength="255" name="address_line_1" placeholder="Adresse" required="" type="text"/></div><div class="form-group"><label class="form-label" for="id_address_line_2">Complément d'adresse</label><input class="form-control" disabled="" id="id_address_line_2" maxlength="255" name="address_line_2" placeholder="Complément d'adresse" type="text"/><div class="form-text">Appartement, suite, bloc, bâtiment, boite postale, etc.</div>
  </div><div class="form-group form-group-required"><label class="form-label" for="id_post_code">Code postal</label><input class="form-control" disabled="" id="id_post_code" maxlength="5" name="post_code" placeholder="Code postal" required="" type="text"/></div><div class="form-group form-group-required"><label class="form-label" for="id_city">Ville</label><input class="form-control" disabled="" id="id_city" maxlength="255" name="city" placeholder="Ville" required="" type="text"/></div><div class="form-group form-group-required"><label class="form-label" for="id_department">Département</label><select class="form-select" disabled="" id="id_department" name="department" required="">
    <option value="">---------</option>
  
    <option value="01">01 - Ain</option>
  
    <option value="02">02 - Aisne</option>
  
    <option value="03">03 - Allier</option>
  
    <option value="04">04 - Alpes-de-Haute-Provence</option>
  
    <option value="05">05 - Hautes-Alpes</option>
  
    <option value="06">06 - Alpes-Maritimes</option>
  
    <option value="07">07 - Ardèche</option>
  
    <option value="08">08 - Ardennes</option>
  
    <option value="09">09 - Ariège</option>
  
    <option value="10">10 - Aube</option>
  
    <option value="11">11 - Aude</option>
  
    <option value="12">12 - Aveyron</option>
  
    <option value="13">13 - Bouches-du-Rhône</option>
  
    <option value="14">14 - Calvados</option>
  
    <option value="15">15 - Cantal</option>
  
    <option value="16">16 - Charente</option>
  
    <option value="17">17 - Charente-Maritime</option>
  
    <option value="18">18 - Cher</option>
  
    <option value="19">19 - Corrèze</option>
  
    <option value="2A">2A - Corse-du-Sud</option>
  
    <option value="2B">2B - Haute-Corse</option>
  
    <option value="21">21 - Côte-d'Or</option>
  
    <option value="22">22 - Côtes-d'Armor</option>
  
    <option value="23">23 - Creuse</option>
  
    <option value="24">24 - Dordogne</option>
  
    <option value="25">25 - Doubs</option>
  
    <option value="26">26 - Drôme</option>
  
    <option value="27">27 - Eure</option>
  
    <option value="28">28 - Eure-et-Loir</option>
  
    <option value="29">29 - Finistère</option>
  
    <option value="30">30 - Gard</option>
  
    <option value="31">31 - Haute-Garonne</option>
  
    <option value="32">32 - Gers</option>
  
    <option value="33">33 - Gironde</option>
  
    <option value="34">34 - Hérault</option>
  
    <option value="35">35 - Ille-et-Vilaine</option>
  
    <option value="36">36 - Indre</option>
  
    <option value="37">37 - Indre-et-Loire</option>
  
    <option value="38">38 - Isère</option>
  
    <option value="39">39 - Jura</option>
  
    <option value="40">40 - Landes</option>
  
    <option value="41">41 - Loir-et-Cher</option>
  
    <option value="42">42 - Loire</option>
  
    <option value="43">43 - Haute-Loire</option>
  
    <option value="44">44 - Loire-Atlantique</option>
  
    <option value="45">45 - Loiret</option>
  
    <option value="46">46 - Lot</option>
  
    <option value="47">47 - Lot-et-Garonne</option>
  
    <option value="48">48 - Lozère</option>
  
    <option value="49">49 - Maine-et-Loire</option>
  
    <option value="50">50 - Manche</option>
  
    <option value="51">51 - Marne</option>
  
    <option value="52">52 - Haute-Marne</option>
  
    <option selected="" value="53">53 - Mayenne</option>
  
    <option value="54">54 - Meurthe-et-Moselle</option>
  
    <option value="55">55 - Meuse</option>
  
    <option value="56">56 - Morbihan</option>
  
    <option value="57">57 - Moselle</option>
  
    <option value="58">58 - Nièvre</option>
  
    <option value="59">59 - Nord</option>
  
    <option value="60">60 - Oise</option>
  
    <option value="61">61 - Orne</option>
  
    <option value="62">62 - Pas-de-Calais</option>
  
    <option value="63">63 - Puy-de-Dôme</option>
  
    <option value="64">64 - Pyrénées-Atlantiques</option>
  
    <option value="65">65 - Hautes-Pyrénées</option>
  
    <option value="66">66 - Pyrénées-Orientales</option>
  
    <option value="67">67 - Bas-Rhin</option>
  
    <option value="68">68 - Haut-Rhin</option>
  
    <option value="69">69 - Rhône</option>
  
    <option value="70">70 - Haute-Saône</option>
  
    <option value="71">71 - Saône-et-Loire</option>
  
    <option value="72">72 - Sarthe</option>
  
    <option value="73">73 - Savoie</option>
  
    <option value="74">74 - Haute-Savoie</option>
  
    <option value="75">75 - Paris</option>
  
    <option value="76">76 - Seine-Maritime</option>
  
    <option value="77">77 - Seine-et-Marne</option>
  
    <option value="78">78 - Yvelines</option>
  
    <option value="79">79 - Deux-Sèvres</option>
  
    <option value="80">80 - Somme</option>
  
    <option value="81">81 - Tarn</option>
  
    <option value="82">82 - Tarn-et-Garonne</option>
  
    <option value="83">83 - Var</option>
  
    <option value="84">84 - Vaucluse</option>
  
    <option value="85">85 - Vendée</option>
  
    <option value="86">86 - Vienne</option>
  
    <option value="87">87 - Haute-Vienne</option>
  
    <option value="88">88 - Vosges</option>
  
    <option value="89">89 - Yonne</option>
  
    <option value="90">90 - Territoire de Belfort</option>
  
    <option value="91">91 - Essonne</option>
  
    <option value="92">92 - Hauts-de-Seine</option>
  
    <option value="93">93 - Seine-Saint-Denis</option>
  
    <option value="94">94 - Val-de-Marne</option>
  
    <option value="95">95 - Val-d'Oise</option>
  
    <option value="971">971 - Guadeloupe</option>
  
    <option value="972">972 - Martinique</option>
  
    <option value="973">973 - Guyane</option>
  
    <option value="974">974 - La Réunion</option>
  
    <option value="975">975 - Saint-Pierre-et-Miquelon</option>
  
    <option value="976">976 - Mayotte</option>
  
    <option value="977">977 - Saint-Barthélémy</option>
  
    <option value="978">978 - Saint-Martin</option>
  
    <option value="986">986 - Wallis-et-Futuna</option>
  
    <option value="987">987 - Polynésie française</option>
  
    <option value="988">988 - Nouvelle-Calédonie</option>
  
  </select></div><div class="form-group"><label class="form-label" for="id_phone">Téléphone</label><input class="form-control" disabled="" id="id_phone" maxlength="20" name="phone" placeholder="Téléphone" type="text" value="0600000000"/></div><div class="form-group"><label class="form-label" for="id_email">E-mail</label><input class="form-control" disabled="" id="id_email" maxlength="254" name="email" placeholder="E-mail" type="email" value="pe@mailinator.com"/></div><div class="form-group"><label class="form-label" for="id_website">Site web</label><input class="form-control" disabled="" id="id_website" maxlength="200" name="website" placeholder="Site web" type="url"/><div class="form-text">Votre site web doit commencer par http:// ou https://</div>
  </div><div class="form-group"><label class="form-label" for="id_description">Description</label><textarea class="form-control" cols="40" disabled="" id="id_description" name="description" placeholder="Description" rows="10"></textarea><div class="form-text">Texte de présentation de votre structure.</div>
  </div>
  
                              
  
  <div class="row">
      <div class="col-12">
          <hr class="mb-3"/>
          <small class="d-inline-block mb-3">* champs obligatoires</small>
          <div class="form-row align-items-center justify-content-end gx-3">
              <div class="form-group col-12 col-lg order-3 order-lg-1">
                  <button aria-label="Annuler la saisie de ce formulaire" class="btn btn-link btn-ico ps-lg-0 w-100 w-lg-auto" data-bs-target="#confirm_reset_modal" data-bs-toggle="modal" type="button">
                      <i aria-hidden="true" class="ri-close-line ri-lg"></i>
                      <span>Annuler</span>
                  </button>
              </div>
              
              <div class="form-group col-12 col-lg-auto order-2 order-lg-3">
                  
                      <button aria-label="Enregistrer la modification" class="btn btn-block btn-primary" type="submit">
                          <span>Enregistrer</span>
                      </button>
                  
              </div>
          </div>
      </div>
  </div>
  
  <div aria-hidden="true" class="modal fade" id="confirm_reset_modal" tabindex="-1">
      <div class="modal-dialog modal-dialog-centered">
          <div class="modal-content">
              <div class="modal-header">
                  <h3 class="modal-title">Êtes-vous sûr de vouloir annuler ?</h3>
              </div>
              <div class="modal-body">
                  Les informations renseignées ne seront pas enregistrées.
                  <br/>
                  Cette action est irréversible.
              </div>
              <div class="modal-footer">
                  <button class="btn btn-sm btn-outline-primary" data-bs-dismiss="modal" type="button">Retour</button>
                  <a class="btn btn-sm btn-danger" href="/dashboard/">Confirmer l'annulation</a>
              </div>
          </div>
      </div>
  </div>
  
                          </form>
  '''
# ---
